driver.go

```diff
@@ -48,13 +48,33 @@ func (l *lvmDriver) Create(req volume.Request) volume.Response {
 		return resp(err)
 	}
 
-	cmdArgs := []string{"-n", req.Name}
+	cmdArgs := []string{"-n", req.Name, "--setactivationskip", "n"}
+	snap, ok := req.Options["snapshot"]
+	isSnapshot := ok && snap != ""
 	s, ok := req.Options["size"]
-	if !ok || (ok && s == "") {
+	hasSize := ok && s != ""
+
+	// no size is valid for thinly-provisioned snapshot volumes
+	if !hasSize && !isSnapshot {
```

Comment:
@shishir-a412ed Yeah, that's a good point. One solution could be to read /etc/lvm/backup/vgname and load some of the information into a struct to get a full view of existing volumes. Docker 1.12 adds a Status field in the response to VolumeDriver.get. It might be useful to set the logical volume type there as well, so that someone could see the type with `docker volume inspect`.

I'd be happy to do a separate pull request to add this.  I don't have much experience with Go, so please critique and give feedback in spots where I'm not writing idiomatic Go or it's just plain wrong.